.github/workflows/go.yml

```diff
@@ -0,0 +1,25 @@
+name: Go
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    types: ["opened", "synchronize", "reopened"]
+
+jobs:
+
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+
+    - name: Set up Go
+      uses: actions/setup-go@v3
+      with:
+        go-version: 1.18
+
```

Comment:
Can we add caching?

- uses: actions/cache@v3
```suggestion

```
        with:
          path: ~/go/pkg/mod
          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
          restore-keys: |
            ${{ runner.os }}-go-
      - name: Setup the project
        run: go mod download

Reply:
@bradbeck disagreed?

Reply:
Oh, I just committed the suggested change, but I guess it wasn't formatted correctly. Will do that in a minute.